In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/sp-500-companies-with-financial-information/financials.csv


## Importing Python libraries for EDA

In [2]:
import numpy as np
import pandas as pd
import seaborn as sns 
import matplotlib.pyplot as plt

## Importing the Dataset

In [3]:
data=pd.read_csv(r"/kaggle/input/sp-500-companies-with-financial-information/financials.csv")
data

Symbol                    Name                  Sector   Price  \
0      MMM              3M Company             Industrials  222.89   
1      AOS         A.O. Smith Corp             Industrials   60.24   
2      ABT     Abbott Laboratories             Health Care   56.27   
3     ABBV             AbbVie Inc.             Health Care  108.48   
4      ACN           Accenture plc  Information Technology  150.51   
..     ...                     ...                     ...     ...   
500    XYL              Xylem Inc.             Industrials   70.24   
501    YUM         Yum! Brands Inc  Consumer Discretionary   76.30   
502    ZBH  Zimmer Biomet Holdings             Health Care  115.53   
503   ZION           Zions Bancorp              Financials   50.71   
504    ZTS                  Zoetis             Health Care   71.51   

     Price/Earnings  Dividend Yield  Earnings/Share  52 Week Low  \
0             24.31        2.332862            7.92       259.77   
1             27.76        1.147959            1.70        68.39   
2             22.51        1.908982            0.26        64.60   
3             19.41        2.499560            3.29       125.86   
4             25.47        1.714470            5.44       162.60   
..              ...             ...             ...          ...   
500           30.94        1.170079            1.83        76.81   
501           27.25        1.797080            4.07        86.93   
502           14.32        0.794834            9.01       133.49   
503           17.73        1.480933            2.60        55.61   
504           32.80        0.682372            1.65        80.13   

     52 Week High    Market Cap        EBITDA  Price/Sales  Price/Book  \
0         175.490  1.387211e+11  9.048000e+09     4.390271       11.34   
1          48.925  1.078342e+10  6.010000e+08     3.575483        6.35   
2          42.280  1.021210e+11  5.744000e+09     3.740480        3.19   
3          60.050  1.813863e+11  1.031000e+10     6.291571       26.14   
4         114.820  9.876586e+10  5.643228e+09     2.604117       10.62   
..            ...           ...           ...          ...         ...   
500        46.860  1.291502e+10  7.220000e+08     2.726209        5.31   
501        62.850  2.700330e+10  2.289000e+09     6.313636      212.08   
502       108.170  2.445470e+10  2.007400e+09     3.164895        2.39   
503        38.430  1.067068e+10  0.000000e+00     3.794579        1.42   
504        52.000  3.599111e+10  1.734000e+09     9.280896       18.09   

                                           SEC Filings  
0    http://www.sec.gov/cgi-bin/browse-edgar?action...  
1    http://www.sec.gov/cgi-bin/browse-edgar?action...  
2    http://www.sec.gov/cgi-bin/browse-edgar?action...  
3    http://www.sec.gov/cgi-bin/browse-edgar?action...  
4    http://www.sec.gov/cgi-bin/browse-edgar?action...  
..                                                 ...  
500  http://www.sec.gov/cgi-bin/browse-edgar?action...  
501  http://www.sec.gov/cgi-bin/browse-edgar?action...  
502  http://www.sec.gov/cgi-bin/browse-edgar?action...  
503  http://www.sec.gov/cgi-bin/browse-edgar?action...  
504  http://www.sec.gov/cgi-bin/browse-edgar?action...  

[505 rows x 14 columns]

## Printing First 10 data

In [4]:
data.head(10)

Symbol                        Name                  Sector   Price  \
0    MMM                  3M Company             Industrials  222.89   
1    AOS             A.O. Smith Corp             Industrials   60.24   
2    ABT         Abbott Laboratories             Health Care   56.27   
3   ABBV                 AbbVie Inc.             Health Care  108.48   
4    ACN               Accenture plc  Information Technology  150.51   
5   ATVI         Activision Blizzard  Information Technology   65.83   
6    AYI           Acuity Brands Inc             Industrials  145.41   
7   ADBE           Adobe Systems Inc  Information Technology  185.16   
8    AAP          Advance Auto Parts  Consumer Discretionary  109.63   
9    AMD  Advanced Micro Devices Inc  Information Technology   11.22   

   Price/Earnings  Dividend Yield  Earnings/Share  52 Week Low  52 Week High  \
0           24.31        2.332862            7.92      259.770       175.490   
1           27.76        1.147959            1.70       68.390        48.925   
2           22.51        1.908982            0.26       64.600        42.280   
3           19.41        2.499560            3.29      125.860        60.050   
4           25.47        1.714470            5.44      162.600       114.820   
5           31.80        0.431903            1.28       74.945        38.930   
6           18.22        0.351185            7.43      225.360       142.000   
7           52.31        0.000000            3.39      204.450       114.451   
8           19.54        0.218321            6.19      169.550        78.810   
9          187.00        0.000000            0.03       15.650         9.700   

     Market Cap        EBITDA  Price/Sales  Price/Book  \
0  1.387211e+11  9.048000e+09     4.390271       11.34   
1  1.078342e+10  6.010000e+08     3.575483        6.35   
2  1.021210e+11  5.744000e+09     3.740480        3.19   
3  1.813863e+11  1.031000e+10     6.291571       26.14   
4  9.876586e+10  5.643228e+09     2.604117       10.62   
5  5.251867e+10  2.704000e+09    10.595120        5.16   
6  6.242378e+09  5.878000e+08     1.795347        3.55   
7  9.455021e+10  2.538040e+09    13.092818       11.06   
8  8.123612e+09  8.539410e+08     1.130106        2.51   
9  1.119166e+10  3.390000e+08     2.109195       21.47   

                                         SEC Filings  
0  http://www.sec.gov/cgi-bin/browse-edgar?action...  
1  http://www.sec.gov/cgi-bin/browse-edgar?action...  
2  http://www.sec.gov/cgi-bin/browse-edgar?action...  
3  http://www.sec.gov/cgi-bin/browse-edgar?action...  
4  http://www.sec.gov/cgi-bin/browse-edgar?action...  
5  http://www.sec.gov/cgi-bin/browse-edgar?action...  
6  http://www.sec.gov/cgi-bin/browse-edgar?action...  
7  http://www.sec.gov/cgi-bin/browse-edgar?action...  
8  http://www.sec.gov/cgi-bin/browse-edgar?action...  
9  http://www.sec.gov/cgi-bin/browse-edgar?action...

## Description of the Dataset - Rows & Columns

In [5]:
data.shape

(505, 14)

* There are 505 rows and 14 columns / features

## Description of the Dataset - Names of Columns

In [6]:
data.columns

Index(['Symbol', 'Name', 'Sector', 'Price', 'Price/Earnings', 'Dividend Yield',
       'Earnings/Share', '52 Week Low', '52 Week High', 'Market Cap', 'EBITDA',
       'Price/Sales', 'Price/Book', 'SEC Filings'],
      dtype='object')

## Description of the Dataset - Datatypes in Columns

In [7]:
data.dtypes

Symbol             object
Name               object
Sector             object
Price             float64
Price/Earnings    float64
Dividend Yield    float64
Earnings/Share    float64
52 Week Low       float64
52 Week High      float64
Market Cap        float64
EBITDA            float64
Price/Sales       float64
Price/Book        float64
SEC Filings        object
dtype: object

* There are 10 numerical columns (Price, Price/Earnings, Dividend Yield, EPS, 52 week Low, 52 week High, Market cap, EBITDA, Price/sales, Price/Book) and one categorical columns (Sector)

## Displaying information about Dataset

* pandas.info() --> prints information about the DataFrame. The information contains the number of columns, column labels, column data types, memory usage, range index, and the number of cells in each column (non-null values).

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 505 entries, 0 to 504
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Symbol          505 non-null    object 
 1   Name            505 non-null    object 
 2   Sector          505 non-null    object 
 3   Price           505 non-null    float64
 4   Price/Earnings  503 non-null    float64
 5   Dividend Yield  505 non-null    float64
 6   Earnings/Share  505 non-null    float64
 7   52 Week Low     505 non-null    float64
 8   52 Week High    505 non-null    float64
 9   Market Cap      505 non-null    float64
 10  EBITDA          505 non-null    float64
 11  Price/Sales     505 non-null    float64
 12  Price/Book      497 non-null    float64
 13  SEC Filings     505 non-null    object 
dtypes: float64(10), object(4)
memory usage: 55.4+ KB


## Printing the description about the dataset

In [9]:
data.describe()

Price  Price/Earnings  Dividend Yield  Earnings/Share  \
count   505.000000      503.000000      505.000000      505.000000   
mean    103.830634       24.808390        1.895953        3.753743   
std     134.427636       41.241081        1.537214        5.689036   
min       2.820000     -251.530000        0.000000      -28.010000   
25%      46.250000       15.350000        0.794834        1.490000   
50%      73.920000       19.450000        1.769255        2.890000   
75%     116.540000       25.750000        2.781114        5.140000   
max    1806.060000      520.150000       12.661196       44.090000   

       52 Week Low  52 Week High    Market Cap        EBITDA  Price/Sales  \
count   505.000000    505.000000  5.050000e+02  5.050000e+02   505.000000   
mean    122.623832     83.536616  4.923944e+10  3.590328e+09     3.941705   
std     155.362140    105.725473  9.005017e+10  6.840544e+09     3.460110   
min       6.590000      2.800000  2.626102e+09 -5.067000e+09     0.153186   
25%      56.250000     38.430000  1.273207e+10  7.739320e+08     1.629490   
50%      86.680000     62.850000  2.140095e+10  1.614399e+09     2.896440   
75%     140.130000     96.660000  4.511968e+10  3.692749e+09     4.703842   
max    2067.990000   1589.000000  8.095080e+11  7.938600e+10    20.094294   

        Price/Book  
count   497.000000  
mean     14.453179  
std      89.660508  
min       0.510000  
25%       2.020000  
50%       3.400000  
75%       6.110000  
max    1403.380000

## Checking Null Values

In [10]:
data.isnull().sum()

Symbol            0
Name              0
Sector            0
Price             0
Price/Earnings    2
Dividend Yield    0
Earnings/Share    0
52 Week Low       0
52 Week High      0
Market Cap        0
EBITDA            0
Price/Sales       0
Price/Book        8
SEC Filings       0
dtype: int64

* There are 2 null values in column Price/Earning(PE ratio) and 8 price/book value

## Printing missing rows in 'Price/Earnings' column

In [11]:
missing_rows_PE = data[data[['Price/Earnings']].isna().any(axis=1)]
print(missing_rows_PE)


    Symbol            Name                  Sector  Price  Price/Earnings  \
360   PRGO         Perrigo             Health Care  84.44             NaN   
430    TPR  Tapestry, Inc.  Consumer Discretionary  48.85             NaN   

     Dividend Yield  Earnings/Share  52 Week Low  52 Week High    Market Cap  \
360        0.731261          -28.01        95.93         63.68  1.232638e+10   
430        0.000000            2.09        50.71         36.69  1.424720e+10   

     EBITDA  Price/Sales  Price/Book  \
360     0.0     3.385007        2.03   
430     0.0     4.078549        4.35   

                                           SEC Filings  
360  http://www.sec.gov/cgi-bin/browse-edgar?action...  
430  http://www.sec.gov/cgi-bin/browse-edgar?action...  


* Row number 360 and 430 has a missing values in 'Price/Earnings' column

## Filling the missing values in 'Price/Earnings' column

In [12]:
data['Price/Earnings'] = data['Price/Earnings'].fillna(data['Price/Earnings'].mean())
print(data.isna().sum())

Symbol            0
Name              0
Sector            0
Price             0
Price/Earnings    0
Dividend Yield    0
Earnings/Share    0
52 Week Low       0
52 Week High      0
Market Cap        0
EBITDA            0
Price/Sales       0
Price/Book        8
SEC Filings       0
dtype: int64


* Now, there is no missing values in 'Price/Book' column

## Printing missing rows in 'Price/Book' column

In [13]:
missing_rows_PB = data[data[['Price/Book']].isna().any(axis=1)]
print(missing_rows_PB)


    Symbol                  Name                  Sector   Price  \
55    ARNC           Arconic Inc             Industrials   24.45   
199     FL       Foot Locker Inc  Consumer Discretionary   45.88   
225    HCA          HCA Holdings             Health Care   95.97   
292    MRO    Marathon Oil Corp.                  Energy   15.68   
346    OXY  Occidental Petroleum                  Energy   68.47   
358    PEP          PepsiCo Inc.        Consumer Staples  110.15   
450    TDG       TransDigm Group             Industrials  283.00   
460    UNP         Union Pacific             Industrials  124.86   

     Price/Earnings  Dividend Yield  Earnings/Share  52 Week Low  \
55            20.21        0.956175           -0.21        31.17   
199            9.50        2.582795            4.91        77.86   
225           14.07        1.422764            5.94       106.84   
292          -32.00        1.224740           -2.65        19.52   
346          195.63        4.408186           -

* Row number: 55,199,225,292,346,358,450,460 has missing values in 'Price/Book' column

## Filling the missing values in 'Price/Book' column

In [14]:
data['Price/Book'] = data['Price/Book'].fillna(data['Price/Book'].mean())
print(data.isna().sum())

Symbol            0
Name              0
Sector            0
Price             0
Price/Earnings    0
Dividend Yield    0
Earnings/Share    0
52 Week Low       0
52 Week High      0
Market Cap        0
EBITDA            0
Price/Sales       0
Price/Book        0
SEC Filings       0
dtype: int64


* Now, there is no missing values in 'Price/Book' column